### ISSUE TITLE
error after putting token (using linux)

### ISSUE THREAD

**Issue by burddan** [AlexFlipnote/GiveMeBadge#13]
Traceback (most recent call last):
  File "aiohttp/connector.py", line 980, in _wrap_create_connection
  File "asyncio/base_events.py", line 1098, in create_connection
  File "asyncio/base_events.py", line 1131, in _create_connection_transport
  File "asyncio/sslproto.py", line 577, in _on_handshake_complete
  File "asyncio/sslproto.py", line 559, in _do_handshake
  File "ssl.py", line 979, in do_handshake
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "index.py", line 74, in <module>
  File "discord/client.py", line 828, in run
  File "asyncio/runners.py", line 190, in run
  File "asyncio/runners.py", line 118, in run
  File "asyncio/base_events.py", line 650, in run_until_complete
  File "discord/client.py", line 817, in runner
  File "discord/client.py", line 745, in start
  File "discord/client.py", line 580, in login
  File "discord/http.py", line 801, in static_login
  File "discord/http.py", line 624, in request
  File "aiohttp/client.py", line 1141, in __aenter__
  File "aiohttp/client.py", line 536, in _request
  File "aiohttp/connector.py", line 540, in connect
  File "aiohttp/connector.py", line 901, in _create_connection
  File "aiohttp/connector.py", line 1206, in _create_direct_connection
  File "aiohttp/connector.py", line 1175, in _create_direct_connection
  File "aiohttp/connector.py", line 982, in _wrap_create_connection
aiohttp.client_exceptions.ClientConnectorCertificateError: Cannot connect to host discord.com:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')]
[5164] Failed to execute script 'index' due to unhandled exception!

**Comment by Husk-ees** [comment]
Same issue for mac as well

**Comment by Nodo08** [comment]
Same issue on MacOS, exact same messages from console.

**Comment by jaden-falls** [comment]
yea

**Comment by Nodo08** [comment]
Update: tried running the entire script on Pycharm, same issue
What worked on windows might not be available directly for macs

**Comment by DarDarDoor** [comment]
Having this issue here as well - Running on fedora linux, python 3.10.7

**Comment by DarDarDoor** [comment]
One very stupid workaround I did find was to run the windows version of this through wine - just using a default wineprefix with the os set to windows 10, everything worked fine. You shouldn't have to do this though, but I guess it can work for now

**Comment by Soft-Bred** [comment]
same issue on macos

**Comment by AlexFlipnote** [comment]
I see a few having the same error, however the majority does not get it. We tested this on Mac, Windows and Linux and got no problems. Are you sure you have a good connection with Discord API? ( https://discord.com/api/v10 )

**Comment by Snowiiii** [comment]
have same error on linux

**Comment by JimmygamerIV** [comment]
I have this error too. I made an account just to say that :) .

**Comment by Snowiiii** [comment]
> I see a few having the same error, however the majority does not get it. We tested this on Mac, Windows and Linux and got no problems. Are you sure you have a good connection with Discord API? ( https://discord.com/api/v10 )

says 404

**Comment by JimmygamerIV** [comment]
Same. It says: {"message": "404: Not Found", "code": 0}

**Comment by DarDarDoor** [comment]
> I see a few having the same error, however the majority does not get it. We tested this on Mac, Windows and Linux and got no problems. Are you sure you have a good connection with Discord API? ( https://discord.com/api/v10 )

Hm, what distro and python version did you test the linux build on? Just curious

**Comment by JimmygamerIV** [comment]
I am on MacOS.

**Comment by Snowiiii** [comment]
Run it in a Windows VM works fine

**Comment by DarDarDoor** [comment]
> Having this issue here as well - Running on fedora linux, python 3.10.7

Alright I just tested this using the source python file instead of the linux build and it worked perfectly. This is an issue with the compiled linux build that's in Releases

**Comment by AlexFlipnote** [comment]
> > I see a few having the same error, however the majority does not get it. We tested this on Mac, Windows and Linux and got no problems. Are you sure you have a good connection with Discord API? ( https://discord.com/api/v10 )
> 
> says 404

Main goal was to see this, sorry for not saying that
![image](https://user-images.githubusercontent.com/19587382/201523384-23c63695-5237-4fd4-86a2-ba7d0dc3df69.png)

**Comment by AlexFlipnote** [comment]
> > Having this issue here as well - Running on fedora linux, python 3.10.7
> 
> Alright I just tested this using the source python file instead of the linux build and it worked perfectly. This is an issue with the compiled linux build that's in Releases

Not sure how tho.. I tested the built version on my Debian 11 server just now and it works fine 🤔

**Comment by AlexFlipnote** [comment]
Linux test screenshot
![image](https://user-images.githubusercontent.com/19587382/201523564-983924df-5b26-40e5-878c-c0f1b322e46a.png)

**Comment by Snowiiii** [comment]
> Linux test screenshot ![image](https://user-images.githubusercontent.com/19587382/201523564-983924df-5b26-40e5-878c-c0f1b322e46a.png)

what python version ?

**Comment by AlexFlipnote** [comment]
It is compiled using Python 3.11, however the application runned is compiled and using binary code, not Python.

**Comment by Snowiiii** [comment]
> It is compiled using Python 3.11.

asyncio have breaking changes in 3.11 its the reason why its not running,But many distros like Arch have 3.10 as current release

**Comment by AlexFlipnote** [comment]
Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?

**Comment by Snowiiii** [comment]
> Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?

you cant change it simply to Python 3.10 because it has breaking changes so you must change some code

**Comment by AlexFlipnote** [comment]
Thing is I do not use anything of asyncio myself, that is done by discord.py themselves

**Comment by DarDarDoor** [comment]
> Could you try the 1.0.4 release just to make sure? https://github.com/AlexFlipnote/GiveMeBadge/releases/tag/1.0.4

Still not working sadly

**Comment by AlexFlipnote** [comment]
Well it was worth a shot (still works for me). I sadly do not control the part of discord.py so.. would need to look into a better solution then.

**Comment by Snowiiii** [comment]
yes not working

**Comment by DarDarDoor** [comment]
> > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> 
> you cant change it simply to Python 3.10 because it has breaking changes so you must change some code

Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly

**Comment by AlexFlipnote** [comment]
> > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > 
> > 
> > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> 
> Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly

When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.

**Comment by Snowiiii** [comment]
> > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > 
> > 
> > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> 
> Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly

what python version do you have ?

**Comment by Nodo08** [comment]
> Could you try the 1.0.4 release just to make sure? https://github.com/AlexFlipnote/GiveMeBadge/releases/tag/1.0.4

Running that in Pycharm sec
3.9 MacOS

**Comment by Nodo08** [comment]
not working still

**Comment by AlexFlipnote** [comment]
> > Could you try the 1.0.4 release just to make sure? https://github.com/AlexFlipnote/GiveMeBadge/releases/tag/1.0.4
> 
> Running that in Pycharm sec 3.9 MacOS

Try this actually:
`pip install discord.py==2.0.1` and run it again

**Comment by DarDarDoor** [comment]
> > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > 
> > > 
> > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > 
> > 
> > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> 
> When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.

Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something

**Comment by AlexFlipnote** [comment]
> > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > 
> > > > 
> > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > 
> > > 
> > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > 
> > 
> > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> 
> Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something

Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure

**Comment by DarDarDoor** [comment]
> > > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > > 
> > > > > 
> > > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > > 
> > > > 
> > > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > > 
> > > 
> > > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> > 
> > 
> > Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something
> 
> Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure

Still not working with the compiled build. Works fine with the .py file

**Comment by AlexFlipnote** [comment]
> > > > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > > > 
> > > > > > 
> > > > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > > > 
> > > > > 
> > > > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > > > 
> > > > 
> > > > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> > > 
> > > 
> > > Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something
> > 
> > 
> > Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure
> 
> Still not working with the compiled build. Works fine with the .py file

python --version ?
Also try doing `make compile` with your Python version and see if that works?

**Comment by AlexFlipnote** [comment]
oh btw, you might need to remove the .exe at end when running the compile from Makefile

**Comment by DarDarDoor** [comment]
> > > > > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > > > > 
> > > > > > > 
> > > > > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > > > > 
> > > > > > 
> > > > > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > > > > 
> > > > > 
> > > > > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> > > > 
> > > > 
> > > > Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something
> > > 
> > > 
> > > Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure
> > 
> > 
> > Still not working with the compiled build. Works fine with the .py file
> 
> python --version ? Also try doing `make compile` with your Python version and see if that works?

Python ver is 3.10.8. It compiled and worked. Finally lol. Still very strange though

**Comment by AlexFlipnote** [comment]
> > > > > > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > > > > > 
> > > > > > > 
> > > > > > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > > > > > 
> > > > > > 
> > > > > > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> > > > > 
> > > > > 
> > > > > Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something
> > > > 
> > > > 
> > > > Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure
> > > 
> > > 
> > > Still not working with the compiled build. Works fine with the .py file
> > 
> > 
> > python --version ? Also try doing `make compile` with your Python version and see if that works?
> 
> Python ver is 3.10.8. It compiled and worked. Finally lol. Still very strange though

Interesting, might it be because I compile using Ubuntu as the docker image option?

**Comment by AlexFlipnote** [comment]
Same result when someone in my server tried too
![image](https://user-images.githubusercontent.com/19587382/201526086-ee48d7cd-3d2a-4c8c-a94f-b538d28a154f.png)

**Comment by Nodo08** [comment]
> > > Could you try the 1.0.4 release just to make sure? https://github.com/AlexFlipnote/GiveMeBadge/releases/tag/1.0.4
> > 
> > 
> > Running that in Pycharm sec 3.9 MacOS
> 
> Try this actually: `pip install discord.py==2.0.1` and run it again

not working..

**Comment by DarDarDoor** [comment]
> > > > > > > > > > Hmm.. I see, let me try changing the compiler to use Python 3.10 and you can test that?
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > you cant change it simply to Python 3.10 because it has breaking changes so you must change some code
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Also what do you mean by this? Like, when you compile it or when you run it? Because i'm still running the src python file through 3.10 and it's working perfectly
> > > > > > > 
> > > > > > > 
> > > > > > > When compiled yes. I use GitHub Workflow to install everything with Python 3.X and then compiles it with that source.
> > > > > > 
> > > > > > 
> > > > > > Very strange. Perhaps github could be messing something up with the build? Probably no way to tell though. Either way there's many different kinds of linux so if you can't find a solution i'd just recommend to get rid of the linux build altogether and just tell people to run the src manually - maybe just simply packaging the python file in a release or something
> > > > > 
> > > > > 
> > > > > Might be the best, however I would still want to test one more thing, since it is the SSL error and not actually being able to run it. Downgrade discord.py to 2.0.1 and see if it works. They recently released 2.1.0 which might have affected it, not 100% sure
> > > > 
> > > > 
> > > > Still not working with the compiled build. Works fine with the .py file
> > > 
> > > 
> > > python --version ? Also try doing `make compile` with your Python version and see if that works?
> > 
> > 
> > Python ver is 3.10.8. It compiled and worked. Finally lol. Still very strange though
> 
> Interesting, might it be because I compile using Ubuntu as the docker image option?

That *might* be the issue, since of course Ubuntu has very outdated packages sometimes. If you want to test my compiled build I just uploaded it here https://anonfiles.com/C2ydoeHfyb/GiveMeBadge - Github doesn't support uploading executables in an issue ofc

**Comment by AlexFlipnote** [comment]
> Just stopping to quote, getting too long lmao

I would prefer if you could fork the project and make a new tag, try compiling it by changing the workflow from using Ubuntu to the distro you use? idk, arch, fedora, etc

**Comment by DarDarDoor** [comment]
> > Just stopping to quote, getting too long lmao (True)
> 
> I would prefer if you could fork the project and make a new tag, try compiling it by changing the workflow from using Ubuntu to the distro you use? idk, arch, fedora, etc

Yeah sure

**Comment by DarDarDoor** [comment]
> > Just stopping to quote, getting too long lmao
> 
> I would prefer if you could fork the project and make a new tag, try compiling it by changing the workflow from using Ubuntu to the distro you use? idk, arch, fedora, etc

Hm well since github doesn't support anything other than ubuntu for actions I'd have to setup my own runner on my own machine which seems like a pain, and it's not something I could always run anyway - apologies

**Comment by AlexFlipnote** [comment]
Shit, thank you for trying at least man, appreciated. I might just consider to tell people who use Linux, to use source instead. Most people who use Linux already knows about this anyways.

**Comment by DarDarDoor** [comment]
> Shit, thank you for trying at least man, appreciated. I might just consider to tell people who use Linux, to use source instead. Most people who use Linux already knows about this anyways.

Yeah, it's alright dw, I can write a new readme with it and open a pr if you'd like. Also did you try my compiled ver?

**Comment by AlexFlipnote** [comment]
```
[18908] Error loading Python lib '/tmp/_MEIwQljnS/libpython3.10.so.1.0': dlopen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /tmp/_MEIwQljnS/libpython3.10.so.1.0)
```

**Comment by AlexFlipnote** [comment]
You could make a PR in README that says it supports Ubuntu/Debian confirmed at least, MacOS not sure from which version and Windows, I think it is win8 and higher. I've had 2-3 people say win7 does not work

**Comment by DarDarDoor** [comment]
> ```
> [18908] Error loading Python lib '/tmp/_MEIwQljnS/libpython3.10.so.1.0': dlopen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /tmp/_MEIwQljnS/libpython3.10.so.1.0)
> ```

Ok, that *might* be because glibc binaries are stored in different places on fedora and debian iirc. Not that it matters anyway though.

**Comment by AlexFlipnote** [comment]
Seems like it does, I kinda thought from the start that Linux might become a problem because distros does different things between them.

**Comment by AlexFlipnote** [comment]
For now I added this to releases
![image](https://user-images.githubusercontent.com/19587382/201528220-657700da-dd3d-4bce-8fbc-82d10418786e.png)

**Comment by DarDarDoor** [comment]
Opened pr https://github.com/AlexFlipnote/GiveMeBadge/pull/19/commits

**Comment by AlexFlipnote** [comment]
Merged #19 that should do for now. Thank you everyone for helping :)
